datalad/interface/rerun.py

```diff
@@ -360,9 +372,41 @@ def _rerun(dset, results):
                 None, ["git", "cherry-pick", res["commit"]],
                 check_fake_dates=True)
             yield res
+        elif rerun_action == "merge":
+            # We uniquify because content may not change (depending on the
+            # value of --onto).
+            new_parents = unique(
+                filter(None, (new_bases.get(p) for p in res["parents"])))
+            if len(new_parents) < 2:
+                lgr.warning(
+                    "Skipping merge %s because only one parent "
+                    "is in the specified revision range",
+                    res["commit"])
+                continue
+            if new_parents[0] != dset.repo.get_hexsha():
+                if branch_to_restore is None:
+                    branch_to_restore = dset.repo.get_active_branch()
+                # Make sure we're merging in the right direction.
+                dset.repo.checkout(new_parents[0])
+            msg = dset.repo.repo.git.show(
+                res["commit"], "--no-patch", "--format=%B")
+            dset.repo._git_custom_command(
+                None,
+                ["git", "merge", "--no-ff", "-m", msg] + new_parents[1:],
```

Comment:
I wonder if it makes sense right away to reserve some options to allow to pass into merge eg to instruct on conflicts resolution strategy

Reply:
A similar comment could be made about cherry picking non-run commits.  I don't plan to do that until the need arises and a convincing case is made.